app/models/restart_signal_handler.rb

```diff
@@ -1,11 +1,18 @@
 # JobQueue locks a mutex, hence the need for a separate SignalHandler thread
 # Self-pipe is also best practice, since signal handlers can themselves be interrupted
-class SignalHandler
+class RestartSignalHandler
+  class << self
+    alias listen new
```

Comment:
for me it was not clear what was happening especially that USR1 was trapped outside and USR2 got raised inside ... so keeping them together made it easier to see this behavior